src/macro_util.rs

```diff
@@ -409,6 +414,83 @@ pub unsafe fn transmute_mut<'dst, 'src: 'dst, Src: 'src, Dst: 'dst>(
     unsafe { &mut *dst }
 }
 
+/// Is a given source a valid instance of `Self`?
+///
+/// # Safety
+///
+/// Unsafe code may assume that, if `is_mut_src_valid(src)` returns true, `*src`
+/// is a bit-valid instance of `Dst`, and that the size of `Src` is greater than
+/// or equal to the size of `Dst`.
+///
+/// # Panics
+///
+/// `is_src_valid` may either produce a post-monomorphization error or a panic
+/// if `Dst` is bigger than `Src`. Otherwise, `is_src_valid` panics under the
+/// same circumstances as [`is_bit_valid`].
+///
+/// [`is_bit_valid`]: TryFromBytes::is_bit_valid
+#[doc(hidden)]
+#[inline]
+fn is_src_valid<Src, Dst, I>(src: Ptr<'_, Src, I>) -> bool
+where
+    Src: IntoBytes,
+    Dst: TryFromBytes,
+    I: Invariants<Validity = invariant::Valid>,
+    I::Aliasing: AtLeast<invariant::Shared>,
+{
+    crate::util::assert_dst_not_bigger_than_src::<Src, Dst>();
+
+    // SAFETY: This is a pointer cast, satisfying the following properties:
+    // - `p as *mut Self` addresses a subset of the `bytes` addressed by
+    //   `c_ptr`, because we assert above that the size of `Self` is less
+    //   than or equal to the size of `Src`.
+    // - `p as *mut Self` is a provenance-preserving cast
+    // - The aliasing of `c_ptr` is neither `Any` nor `Shared`, so we do not
+    //   need to reason about the ranges of `UnsafeCell`s.
```

Comment:
Looks like these were copy-pasted without being updated?

Reply:
Not just that, but a soundness issue too! Fixed, thanks to our spiffy `AliasingSafe` trait.